lab/dp/problems/sum_query/main.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 Nichita Radu <radunichita99@gmail.com>
+
+// SumQuery - Programare dinamica
+// Complexitate O(M * N)
+
+// https://leetcode.com/problems/range-sum-query-2d-immutable/
+
+#include <bits/stdc++.h>
+
+class Task {
+    // N si M sunt dimensiunile matricei, K e numarul de query-uri
+    int N, M, K;
+
+    // matricea de elemente
+    std::vector<std::vector<int>> matrix;
+
+    // query-urile de forma <(start_x, start,y), (finish_x), (finish_y)>.
+    std::vector<std::pair<std::pair<int, int>, std::pair<int, int>>> queries;
+
+    // pe pozitia [i][j] am suma elementelor submatricei (1,1) pana la (i, j).
+    std::vector<std::vector<int>> dp;
+
+    // vectorul se solutii
+    std::vector<int> solution;
+
+    void read_input() {
+        std::ifstream in("sum_query.in");
+        in >> N;
+        in >> M;
+        in >> K;
+        matrix =
+        std::vector<std::vector<int>>(N + 1, std::vector<int>(M + 1, 0));
+        dp =
```

Comment:
Did you run clang-format? I bet on not!

Reply:
I had some trouble, but a purge and a clean install did the trick as always :)